### PR TITLE
libc/stdio: support hh, h, int, long, long long modifier in sscanf

### PR DESCRIPTION
**Pull request by sunghan-chang**
Currenlty it does not support. When the type of getting result
is less than int, sscanf removes outsid of result buffer.
for example,
char ret[10] = "efghi";
char buf[10] = "1:2:3";

sscanf(buf, "%hhx:%hhx:%hhx", &ret[0], &ret[1], &ret[2]);

Then, ret should be "123hi", but because of not support hh modifier
it becomes "123".

This is partially back-ported from nuttx commit 350295d.

Signed-off-by: sunghan-chang <sh924.chang@samsung.com>

### CODE REVIEW

File: lib/libc/stdio/lib_sscanf.c [hunk 1]
```diff
@@ -830,19 +909,38 @@ int vsscanf(FAR const char *buf, FAR const char *fmt, va_list ap)
...
+						pchar = va_arg(ap, unsigned char *);
+						*pchar = (unsigned char)nchars;
+						break;
+					case H_MOD:
+						pshort = va_arg(ap, unsigned short *);
+						*pshort = (unsigned short)nchars;
+						break;
+					case NO_MOD:
+						pint = va_arg(ap, unsigned int *);
+						*pint = (unsigned int)nchars;;
```
Comment (jeongchanKim) [inline lib/libc/stdio/lib_sscanf.c:923]: Please remove unnecessary `;`.

File: lib/libc/stdio/lib_sscanf.c [hunk 2]
```diff
@@ -340,13 +346,20 @@ int vsscanf(FAR const char *buf, FAR const char *fmt, va_list ap)
 	FAR const char *bufstart;
 	FAR char *tv;
 	FAR const char *tc;
-	bool lflag;
+	int modifier;
 	bool noassign;
 	int count;
 	int width;
 	int fwidth;
 	int base = 10;
```
Comment (jsdosa) [inline lib/libc/stdio/lib_sscanf.c:354]: in my opinion, It seems better to add a definition for 10.

Comment (sunghan-chang) [reply lib/libc/stdio/lib_sscanf.c:354]: This is not related with this PR. Let's do that later with new PR.

Comment (jsdosa) [reply lib/libc/stdio/lib_sscanf.c:354]: ok, i see.

**Pull request merged**